openslides_backend/action/actions/projector/project.py

```diff
@@ -0,0 +1,119 @@
+from typing import Any, Dict
+
+from ....models.models import Projection, Projector
+from ....shared.filters import And, FilterOperator, Not
+from ....shared.patterns import Collection, string_to_fqid
+from ....shared.schema import required_id_schema
+from ...generics.update import UpdateAction
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionData
+from ..projection.create import ProjectionCreate
+from ..projection.update import ProjectionUpdate
+
+
+@register_action("projector.project")
+class ProjectorProject(UpdateAction):
+    """
+    Action to projector project.
+    """
+
+    model = Projector()
+    schema = DefaultSchema(Projection()).get_default_schema(
+        required_properties=["content_object_id"],
+        optional_properties=["options", "stable", "type"],
+        additional_required_fields={
+            "ids": {"type": "array", "items": required_id_schema}
+        },
+        title="Projector project schema",
+    )
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        for instance in action_data:
+            content_object = self.datastore.get(
+                string_to_fqid(instance["content_object_id"]), ["meeting_id"]
+            )
+            meeting_id = content_object["meeting_id"]
+
+            if instance.get("stable", False) is False:
+                self.move_equal_projections_to_history(instance, meeting_id)
+                self.move_unstable_projections_to_history(instance)
+
+            create_data = []
+            for projector_id in instance["ids"]:
+                data_dict = {
+                    "current_projector_id": projector_id,
+                    "meeting_id": meeting_id,
+                }
+                for field in ("content_object_id", "options", "stable", "type"):
```

Comment:
`content_object_id` can directly given in `data_dict`.

Reply:
Okay, changed.